android/java/org/chromium/chrome/browser/crypto_wallet/fragments/RecoveryPhraseSaveCopyBottomSheetDialogFragment.java

```diff
@@ -0,0 +1,134 @@
+    LinearLayout mCopyLayout;
+    LinearLayout mSaveLayout;
+    private String mRecoveryPhrasesText;
+
+    public static RecoveryPhraseSaveCopyBottomSheetDialogFragment newInstance(String recoveryPhrasesText) {
+        RecoveryPhraseSaveCopyBottomSheetDialogFragment recoveryPhraseSaveCopyBottomSheetDialogFragment =
+                new RecoveryPhraseSaveCopyBottomSheetDialogFragment();
+        Bundle args = new Bundle();
+        args.putCharSequence("recoveryPhrasesText", recoveryPhrasesText);
+        recoveryPhraseSaveCopyBottomSheetDialogFragment.setArguments(args);
+        return recoveryPhraseSaveCopyBottomSheetDialogFragment;
+    }
+
+    @Override
+    public void show(FragmentManager manager, String tag) {
+        try {
+            RecoveryPhraseSaveCopyBottomSheetDialogFragment fragment =
+                    (RecoveryPhraseSaveCopyBottomSheetDialogFragment) manager.findFragmentByTag(
+                            RecoveryPhraseSaveCopyBottomSheetDialogFragment.TAG_FRAGMENT);
+            FragmentTransaction transaction = manager.beginTransaction();
+            if (fragment != null) {
+                transaction.remove(fragment);
+            }
+            transaction.add(this, tag);
+            transaction.commitAllowingStateLoss();
+        } catch (IllegalStateException e) {
+            Log.e("RecoveryPhraseSaveCopyBottomSheetDialogFragment", e.getMessage());
+        }
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mRecoveryPhrasesText = getArguments().getCharSequence("recoveryPhrasesText").toString();
+    }
+
+    @Override
+    public void setupDialog(@NonNull Dialog dialog, int style) {
+        super.setupDialog(dialog, style);
+
+        @SuppressLint("InflateParams")
+        final View view =
+                LayoutInflater.from(getContext()).inflate(R.layout.copy_save_bottom_sheet, null);
+
+        mCopyLayout = view.findViewById(R.id.copy_layout);
+        mCopyLayout.setOnClickListener(this);
+        mSaveLayout = view.findViewById(R.id.save_layout);
+        mSaveLayout.setOnClickListener(this);
+
+        dialog.setContentView(view);
+        ViewParent parent = view.getParent();
+        ((View) parent).getLayoutParams().height = ViewGroup.LayoutParams.WRAP_CONTENT;
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view == mSaveLayout) {
+            Utils.LaunchBackupFilePicker((AsyncInitializationActivity) getActivity(), this);
+        } else if (view == mCopyLayout) {
+            Utils.saveTextToClipboard(getActivity(), mRecoveryPhrasesText);
+        }
+        dismiss();
+    }
+
+    @Override
+    public void onIntentCompleted(int resultCode, Intent results) {
+        if (resultCode != Activity.RESULT_OK || results == null || results.getData() == null) {
+            Utils.onFileNotSelected();
+            return;
+        }
+
+        new AsyncTask<Void>() {
```

Comment:
`AsyncTask` is deprecated, we should use `java.util.concurrent`. See my original comment.